src/components/ha-icon.ts

```diff
@@ -1,37 +1,113 @@
-import { Constructor } from "../types";
+import {
+  LitElement,
+  customElement,
+  CSSResult,
+  css,
+  svg,
+  property,
+  PropertyValues,
+} from "lit-element";
 
-import "@polymer/iron-icon/iron-icon";
-// Not duplicate, this is for typing.
-// tslint:disable-next-line
-import { IronIconElement } from "@polymer/iron-icon/iron-icon";
+import { get, Store } from "idb-keyval";
 
-const ironIconClass = customElements.get("iron-icon") as Constructor<
-  IronIconElement
->;
+const iconStore = new Store("hass-icon-db", "mdi-icon-store");
+const partPromise = {};
 
-let loaded = false;
+@customElement("ha-icon")
+export class HaIcon extends LitElement {
+  @property() public icon?: string;
+  @property() public path?: string;
```

Comment:
Let's not have 1 component do both. There is no need. Let's create 1 `HaSvgIcon` component and have `HaIcon` use that to render the icon.